rskj-core/src/main/java/co/rsk/peg/federation/FederationTracker.java

```diff
@@ -0,0 +1,27 @@
+package co.rsk.peg.federation;
+
+public abstract class FederationTracker<T> {
```

Comment:
Not sure if we could use an observer pattern here. Would it be possible as things are now? Would it be too much work? This way we avoid using the `modified` flag.

Reply:
I think before tacking this we need to understand the modified flag behavior. Is it only here as an optimization 	that prevents redundant repository calls when the state is accessed multiple times? Is it due to high gas costs? Can we just do a simple read-write strategy? Then we can see how to approach this, but ideally we do not want to add modified flags or any unnecessary code.

Reply:
agree

Reply:
Very interesting discussion. Agree with @apancorb